graphqlapi/src/main/java/io/stargate/graphql/persistence/schemafirst/SchemaSourceDao.java

```diff
@@ -0,0 +1,311 @@
+    } else {
+      resultSet = dataStore.execute(schemaQuery(namespace)).get();
+    }
+    if (!resultSet.iterator().hasNext()) {
+      return null;
+    }
+    return toSchemaSource(namespace, resultSet.one());
+  }
+
+  public SchemaSource getLatest(String namespace) throws Exception {
+    return getByVersion(namespace, Optional.empty());
+  }
+
+  private boolean tableExists() {
+    Keyspace keyspace = dataStore.schema().keyspace(KEYSPACE_NAME);
+    if (keyspace == null) {
+      return false;
+    }
+    Table table = keyspace.table(TABLE_NAME);
+    if (table == null) {
+      return false;
+    }
+    failIfUnexpectedSchema(table);
+    return true;
+  }
+
+  private SchemaSource toSchemaSource(String namespace, Row r) {
+    return new SchemaSource(
+        namespace, r.getUuid(VERSION_COLUMN_NAME), r.getString(CONTENTS_COLUMN_NAME));
+  }
+
+  @VisibleForTesting
+  BoundQuery schemaQueryWithSpecificVersion(String namespace, UUID uuid) {
+    return dataStore
+        .queryBuilder()
+        .select()
+        .column(VERSION_COLUMN_NAME, CONTENTS_COLUMN_NAME)
+        .from(KEYSPACE_NAME, TABLE_NAME)
+        .where(NAMESPACE_COLUMN_NAME, Predicate.EQ, namespace)
+        .where(VERSION_COLUMN_NAME, Predicate.EQ, uuid)
+        .build()
+        .bind();
+  }
+
+  @VisibleForTesting
+  BoundQuery schemaQuery(String namespace) {
+    return dataStore
+        .queryBuilder()
+        .select()
+        .column(VERSION_COLUMN_NAME, CONTENTS_COLUMN_NAME)
+        .from(KEYSPACE_NAME, TABLE_NAME)
+        .where(NAMESPACE_COLUMN_NAME, Predicate.EQ, namespace)
+        .orderBy(VERSION_COLUMN_NAME, Column.Order.DESC)
+        .build()
+        .bind();
+  }
+
+  /** @return the new version */
+  public SchemaSource insert(String namespace, String newContents) {
+
+    UUID newVersion = Uuids.timeBased();
+
+    BoundQuery insertNewSchema =
+        dataStore
+            .queryBuilder()
+            .insertInto(KEYSPACE_NAME, TABLE_NAME)
+            .value(NAMESPACE_COLUMN_NAME, namespace)
+            .value(VERSION_COLUMN_NAME, newVersion)
+            .value(LATEST_VERSION_COLUMN_NAME, newVersion)
+            .value(CONTENTS_COLUMN_NAME, newContents)
+            .value(DEPLOYMENT_IN_PROGRESS_COLUMN_NAME, false)
+            .build()
+            .bind();
+
+    try {
+      dataStore.execute(insertNewSchema).get();
+    } catch (Exception e) {
+      throw new RuntimeException(
+          String.format(
+              "Schema deployment for namespace: %s and version: %s failed.",
+              namespace, newVersion));
+    }
+    return new SchemaSource(namespace, newVersion, newContents);
+  }
+
+  private void ensureTableExists() throws Exception {
+    dataStore
+        .execute(
+            dataStore
+                .queryBuilder()
+                .create()
+                .keyspace(KEYSPACE_NAME)
+                .ifNotExists()
+                .withReplication(Replication.simpleStrategy(1))
```

Comment:
Could this be flaky is realistic replicated environments? With `RF==1` if one of the storage nodes is taken down for maintenance, this keyspace will become unavailable to a portion of clients :thinking:

Reply:
Yeah I think we're going to want to make this configurable.

Reply:
Plus wrap it in a flag for whether or not SG should be allowed to create the keyspace and table on the fly.

Reply:
Yes, this is the default if the keyspace gets created by the first deployment, and we have to use a common denominator because we don't know the cluster topology. Ideally it should be created ahead of time by an administrator with a higher RF, I'll make sure to mention it in the docs.